src/hotspot/os_cpu/linux_aarch64/os_linux_aarch64.cpp

```diff
@@ -382,7 +382,10 @@ int os::extra_bang_size_in_bytes() {
 
 extern "C" {
   int SpinPause() {
-    return 0;
+    // ISB creates a small delay without consuming ALU resources.
+    // This allows it to act as x86 PAUSE.
+    __asm volatile("isb");
```

Comment:
I hadn't realised this was unconditional. Similar to my comment on the onSpinWait patch, while the ISB instruction happens to work for you just now on the platforms you have tested, there is no guarantee that it will continue to do so in future processors. 
Could this be made optional? This is awkward, as it is C++ code, but it may have to be a build configuration option.

Reply:
Hi Stuart,
Yes, I'll be updating it to support different implementations.